ihmc-common-walking-control-modules/src/test/java/us/ihmc/commonWalkingControlModules/capturePoint/splitFractionCalculation/SplitFractionToolsTest.java

```diff
@@ -0,0 +1,162 @@
+   private double transferSplitFraction = 0.0;
+   private double actualSplitFraction = 0.0;
+   private double expectedForwardShift = 0.0;
+   private double expectedBackwardShift = 0.0;
+
+   public void calculateValues()
+   {
+      double percentShift = 0.0;
+      if (transferSplitFraction > defaultSplitFraction)
+      {
+         percentShift = Math.abs(transferSplitFraction - defaultSplitFraction) / (1 - defaultSplitFraction);
+      }
+      else
+      {
+         percentShift = Math.abs(transferSplitFraction - defaultSplitFraction) / defaultSplitFraction;
+      }
+
+      actualSplitFraction = SplitFractionTools.appendSplitFraction(transferSplitFraction, currentSplitFraction, defaultSplitFraction);
+      double expectedForwardShiftedSplitFraction = percentShift * (1 - currentSplitFraction);
+      double expectedBackwardShiftedSplitFraction = percentShift * currentSplitFraction;
+      expectedForwardShift = currentSplitFraction + expectedForwardShiftedSplitFraction;
+      expectedBackwardShift = currentSplitFraction - expectedBackwardShiftedSplitFraction;
+   }
+
+   @Test
+   public void testEqual()
+   {
+      defaultSplitFraction = 0.5;
+      currentSplitFraction = 0.5;
+      transferSplitFraction = 0.5;
+      calculateValues();
+      assertEquals(expectedForwardShift, actualSplitFraction, epsilon);
+   }
+
+   @Test
+   public void testForward()
+   {
+      defaultSplitFraction = 0.5;
+      currentSplitFraction = 0.5;
+      transferSplitFraction = 0.7;
+      calculateValues();
+      assertEquals(expectedForwardShift, actualSplitFraction, epsilon);
+   }
+
+   @Test
+   public void testBackward()
+   {
+      defaultSplitFraction = 0.7;
+      currentSplitFraction = 0.5;
+      transferSplitFraction = 0.3;
+      calculateValues();
+      assertEquals(expectedBackwardShift, actualSplitFraction, epsilon);
+   }
+
+   @Test
+   public void testNegative()
+   {
+      defaultSplitFraction = -0.7;
+      currentSplitFraction = -0.5;
+      transferSplitFraction = -0.3;
+      defaultSplitFraction = clamp(defaultSplitFraction, 0, 1);
+      currentSplitFraction = clamp(currentSplitFraction, 0, 1);
+      transferSplitFraction = clamp(transferSplitFraction, 0, 1);
+      calculateValues();
+      assertEquals(expectedBackwardShift, actualSplitFraction, epsilon);
+   }
+
+   @Test
+   public void testRandom()
+   {
+      Random random = new Random(66);
+      defaultSplitFraction = random.nextDouble();
+      currentSplitFraction = random.nextDouble();
+      transferSplitFraction = random.nextDouble();
+      System.out.printf("Values: %.2f, %.2f, %.2f\n", defaultSplitFraction, currentSplitFraction, transferSplitFraction);
+      calculateValues();
+      if (transferSplitFraction > defaultSplitFraction)
+      {
+         assertEquals(expectedForwardShift, actualSplitFraction, epsilon);
+      }
+      else
+      {
+         assertEquals(expectedBackwardShift, actualSplitFraction, epsilon);
+      }
```

Comment:
This is a good start. However, random will only return values that are between 0 and 1. do you want to test values that are out of bounds? 

Also we should wrap this in a for loop. Testing one random double instance is good, testing 1e5 instances is better.